omod/src/main/webapp/resources/scripts/custom/deletePatient.js

```diff
@@ -0,0 +1,66 @@
+var delPatient = delPatient || {};
+var jqObject = $();
+
+delPatient.deletePatientCreationDialog = null;
+
+/**
+ * Functions used to Delete a Patient
+ */
+
+delPatient.returnUrl = "/coreapps/findpatient/findPatient.page?app=coreapps.findPatient";
+
+delPatient.createDeletePatientCreationDialog = function() {
+    delPatient.deletePatientCreationDialog = emr.setupConfirmationDialog({
+        selector: '#delete-patient-creation-dialog',
+        actions: {
+            confirm: function() {
+                var deleteReason = jQuery('#delete-reason').val().trim(); //Retrieve text from text box
+                if(deleteReason && deleteReason.length > 0) { //Should not be invalid or empty
+                    jq.ajax({
+                        url: '/openmrs/ws/rest/v1/patient/'+delPatient.patientUUID,
+                        type: 'DELETE',
+                        body: {
+                            "void_reason": deleteReason
+                        },
+                        success: function() {
+                            jqObject.toastmessage( 'showToast', { type: 'success',
+                                position: 'top-right',
+                                text:  emr.successMessage("Success!") } );
+                            jq('#delete-patient-creation-dialog' + ' .icon-spin').css('display', 'inline-block').parent().addClass('disabled');
+                            delPatient.deletePatientCreationDialog.close();
+                            jq(setTimeout(delPatient.goToReturnUrl, 1300)); //Allow the success message to display before redirecting to another page
+                        },
+                        error: function() {
+                            jqObject.toastmessage( 'showToast', { type: 'error',
```

Comment:
I honestly did not know that. I'll look into it

On Fri, May 6, 2016 at 12:01 AM, Wyclif Luyima <notifications@github.com>
wrote:

> In omod/src/main/webapp/resources/scripts/custom/deletePatient.js
> <https://github.com/openmrs/openmrs-module-coreapps/pull/20#discussion_r62232106>
> :
>
> > +                    jq.ajax({
> > +                        url: '/openmrs/ws/rest/v1/patient/'+delPatient.patientUUID,
> > +                        type: 'DELETE',
> > +                        body: {
> > +                            "void_reason": deleteReason
> > +                        },
> > +                        success: function() {
> > +                            jqObject.toastmessage( 'showToast', { type: 'success',
> > +                                position: 'top-right',
> > +                                text:  emr.successMessage("Success!") } );
> > +                            jq('#delete-patient-creation-dialog' + ' .icon-spin').css('display', 'inline-block').parent().addClass('disabled');
> > +                            delPatient.deletePatientCreationDialog.close();
> > +                            jq(setTimeout(delPatient.goToReturnUrl, 1300)); //Allow the success message to display before redirecting to another page
> > +                        },
> > +                        error: function() {
> > +                            jqObject.toastmessage( 'showToast', { type: 'error',
>
> emr.loadMessages
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-module-coreapps/pull/20/files/41ec73784d74d4ab1d46df69219e7d26689ba6fe#r62232106>
>